x-pack/plugins/maps/public/shared/layers/sources/es_join_source.js

```diff
@@ -107,13 +112,13 @@ export class ESJoinSource extends AbstractESSource {
       searchSource.setField('index', indexPattern);
       searchSource.setField('size', 0);
       searchSource.setField('filter', () => {
-        const filters = [];
+        const allFilters = [...filters];
```

Comment:
This is an interesting bug that the filters were not getting applied to join search. I will open a separate PR that updates es_join_source to use AbstractESSource._makeSearchSource and remove this duplication/bug.

Reply:
Created https://github.com/elastic/kibana/pull/34072 to address this issue